web/src/app/selection/MaterialSelect.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useState, ReactNode, ReactElement, ChangeEvent } from 'react'
+import {
+  TextField,
+  makeStyles,
+  MenuItem,
+  ListItemIcon,
+  Typography,
+  Paper,
+  Chip,
+} from '@material-ui/core'
+import { Autocomplete } from '@material-ui/lab'
+
+function asArray(value: any) {
+  if (!value) return []
+
+  return Array.isArray(value) ? value : [value]
+}
+
+const useStyles = makeStyles({
+  listItemIcon: {
+    minWidth: 0,
+  },
+  menuItem: {
+    display: 'flex',
+    flex: 1,
+    justifyContent: 'space-between',
+    wordBreak: 'break-word',
+    whiteSpace: 'pre-wrap',
+  },
+  option: {
+    padding: 0,
+  },
+  clearIndicator: {
+    display: 'none',
+  },
+})
+
+interface SelectOption {
+  icon?: ReactElement
+  isCreate?: boolean
+  label: string
+  value: string
+}
+
+interface MaterialSelectProps {
+  disabled?: boolean
+  error?: boolean
+  isLoading?: boolean
+  label?: string
+  multiple?: boolean // allow selecting multiple values
+  noOptionsText?: ReactNode
+  name?: string
+  required?: boolean
+  onChange: Function
```

Comment:
Let's define the signatures here instead of just using `Function`

Something like:
```ts
onChange: (value: string|string[]) => void
onInputChange: (value: string) => void
```

Reply:
Think we could swap props using `multiple`?

```ts
interface MaterialSelectCommonProps {
    disabled?: boolean
    // shared stuff
}
interface MaterialSelectSingleProps extends MaterialSelectCommonProps {
    multiple?: false
    value?: SelectOption
    onChange: (SelectOption) => void
}
interface MaterialSelectMultiProps extends MaterialSelectCommonProps {
    multiple: true
    value?: SelectOption[]
    onChange: (SelectOption[]) => void
}
```